frontend/ui/widget/checkbutton.lua

```diff
@@ -109,39 +109,39 @@ function CheckButton:initCheckButton(checked)
 end
 
 function CheckButton:onTapCheckButton()
-    if self.enabled and self.callback then
-        if G_reader_settings:isFalse("flash_ui") then
-            self.callback()
+    if self.enabled then
+        if self.tap_input then
+            self:onInput(self.tap_input)
+        elseif type(self.tap_input_func) == "function" then
+            self:onInput(self.tap_input_func())
         else
-            -- c.f., ui/widget/iconbutton for the canonical documentation about the flash_ui code flow
-
-            local highlight_dimen = self.dimen
-
-            -- Highlight
-            --
-            self[1].invert = true
-            UIManager:widgetInvert(self[1], highlight_dimen.x, highlight_dimen.y, highlight_dimen.w)
-            UIManager:setDirty(nil, "fast", highlight_dimen)
-
-            UIManager:forceRePaint()
-            UIManager:yieldToEPDC()
-
-            -- Unhighlight
-            --
-            self[1].invert = false
-            UIManager:widgetInvert(self[1], highlight_dimen.x, highlight_dimen.y, highlight_dimen.w)
-            UIManager:setDirty(nil, "ui", highlight_dimen)
-
-            -- Callback
-            --
-            self.callback()
-
-            UIManager:forceRePaint()
```

Comment:
This ordering is changed now, any reason except deduplicating self.callback() ?
All this is from @NiLuJe flashing mind, so I guess he should confirm this feels ok.

Reply:
Restored